src/qt/consolidateunspentwizardselectinputspage.cpp

```diff
@@ -0,0 +1,719 @@
+                                                                         COLUMN_PRIORITY : sortColumn)),
+                                               sortOrder);
+}
+
+// treeview: clicked on header
+void ConsolidateUnspentWizardSelectInputsPage::headerSectionClicked(int logicalIndex)
+{
+    if (logicalIndex == COLUMN_CHECKBOX) // click on most left column -> do nothing
+    {
+        ui->treeWidget->header()->setSortIndicator((sortColumn == COLUMN_AMOUNT_INT64 ?
+                                                        COLUMN_AMOUNT : (sortColumn == COLUMN_PRIORITY_INT64 ?
+                                                                             COLUMN_PRIORITY : sortColumn)),
+                                                   sortOrder);
+    }
+    else
+    {
+        if (logicalIndex == COLUMN_AMOUNT) // sort by amount
+            logicalIndex = COLUMN_AMOUNT_INT64;
+
+        if (logicalIndex == COLUMN_PRIORITY) // sort by priority
+            logicalIndex = COLUMN_PRIORITY_INT64;
+
+        if (sortColumn == logicalIndex)
+            sortOrder = ((sortOrder == Qt::AscendingOrder) ? Qt::DescendingOrder : Qt::AscendingOrder);
+        else
+        {
+            sortColumn = logicalIndex;
+
+            // if amount,date,conf,priority then default => desc, else default => asc
+            sortOrder = ((sortColumn == COLUMN_AMOUNT_INT64 || sortColumn == COLUMN_PRIORITY_INT64
+                          || sortColumn == COLUMN_DATE || sortColumn == COLUMN_CONFIRMATIONS) ?
+                             Qt::DescendingOrder : Qt::AscendingOrder);
+        }
+
+        sortView(sortColumn, sortOrder);
+    }
+}
+
+
+// toggle tree mode
+void ConsolidateUnspentWizardSelectInputsPage::treeModeRadioButton(bool checked)
+{
+    if (checked && model)
+        updateView();
+}
+
+// toggle list mode
+void ConsolidateUnspentWizardSelectInputsPage::listModeRadioButton(bool checked)
+{
+    if (checked && model)
+        updateView();
+}
+
+// checkbox clicked by user
+void ConsolidateUnspentWizardSelectInputsPage::viewItemChanged(QTreeWidgetItem* item, int column)
+{
+    if (!m_ViewItemsChangedViaFilter) m_InputSelectionLimitedByFilter = false;
+
+    if (column == COLUMN_CHECKBOX)
+    {
+        // transaction hash is 64 characters (this means its a child node, so its not a parent node in tree mode)
+        if (item->text(COLUMN_TXHASH).length() == 64)
+        {
+            COutPoint outpt(uint256S(item->text(COLUMN_TXHASH).toStdString()), item->text(COLUMN_VOUT_INDEX).toUInt());
+
+            if (item->checkState(COLUMN_CHECKBOX) == Qt::Unchecked)
+            {
+                coinControl->UnSelect(outpt);
+            }
+            else if (item->isDisabled()) // locked (this happens if "check all" through parent node)
+            {
+                item->setCheckState(COLUMN_CHECKBOX, Qt::Unchecked);
+            }
+            else
+            {
+                coinControl->Select(outpt);
```

Comment:
It is the same object since the CCoinControl object is passed in by pointer:

void ConsolidateUnspentWizardSelectInputsPage::setCoinControl(CCoinControl *coinControl)
{
    this->coinControl = coinControl;
}

Proof this does not happen as you described:

Start state:
![image](https://user-images.githubusercontent.com/7529186/118016385-52d74900-b323-11eb-93d7-37af526dae0d.png)

Push consolidate wizard button:
![image](https://user-images.githubusercontent.com/7529186/118016468-6da9bd80-b323-11eb-83a3-4a5bd5e5d969.png)

Select two inputs:
![image](https://user-images.githubusercontent.com/7529186/118016599-9762e480-b323-11eb-87ab-f24deb75f79f.png)

Cancel out of consolidate wizard and push coin control button:
![image](https://user-images.githubusercontent.com/7529186/118017098-325bbe80-b324-11eb-98b9-a0c744d2072d.png)

So the coin control dialog sees it. When you ok the coin control or press X on coincontrol, the labels update properly.
![image](https://user-images.githubusercontent.com/7529186/118017252-633bf380-b324-11eb-9e16-2ee402c6ca9c.png)

Maybe I need to put another signal in back to the send form?

Reply:
A signal emitted when the wizard closes could work I think, or the wizard could use a temporary coin control object that passes back to the send form when the wizard finishes to insulate the working changes from the main form.

In the screenshots above, the issue occurs after pressing "cancel" in the wizard if the user never pushes the coin control button for the next step. The labels don't reflect the coin control state applied to the transaction.

Reply:
I would rather not be passing around temp coinControl objects. I think a signal is the best approach. Let me try that.

Reply:
It was even simpler than that. Just missing a coinControlUpdateLabels(); after the .exec().